extension/delta/src/connector/delta_connector.cpp

```diff
@@ -10,8 +10,14 @@ void DeltaConnector::connect(const std::string& /*dbPath*/, const std::string& /
     // Creates an in-memory duckdb instance, then install httpfs and attach postgres.
     instance = std::make_unique<duckdb::DuckDB>(nullptr);
     connection = std::make_unique<duckdb::Connection>(*instance);
-    executeQuery("install delta;");
-    executeQuery("load delta;");
+    // Install the Desired Extension on DuckDB
+    if (format == "DELTA") {
```

Comment:
i would create a new connector for iceberg instead of reusing the delta connector.
The connector code is also quite short